unit_tests/test_decode_backtrace.py

```diff
@@ -121,21 +135,149 @@ def test_04_decode_backtraces_core(self):
+        # Set MAX_COREDUMP_THREAD_EXCEPTIONS to 1 for decrease time of running test
+        exceptions_amount = 2
+        self.node.max_coredump_thread_exceptions = 1
+        self.node.remoter.test_exception = SSHConnectTimeoutError
+        self.node.remoter.exceptions_amount = exceptions_amount
+
+        self.node.start_backtrace_thread()
+
+        print("Wait for all exceptions raised")
+        wait.wait_for(self.wait_for_all_exceptions_raised, timeout=50, exceptions_amount=exceptions_amount,
+                      text="Wait for all exceptions raised")
+
+        print("Wait for raised exception found")
+        traceback = wait.wait_for(self.find_last_event, step=5, timeout=10)
+
+        assert "sdcm.remote.base.SSHConnectTimeoutError" == traceback, \
+               f"Expected exception 'sdcm.remote.base.SSHConnectTimeoutError', got {traceback}"
+
+    def test_backtrace_thread_with_any_exception(self):
+        """
+        Test case when get backtraces function failed after an exception was raised 3 times.
+        Expected that coredump thread will be stopped after 3 exceptions and last exception will be raised.
+        """
+
+        # In case when non network exception is thrown during getting backtrace,
+        # exception will be raised from the coredump thread after MAX_COREDUMP_THREAD_EXCEPTIONS attempts.
+        # Set MAX_COREDUMP_THREAD_EXCEPTIONS to 3 for decrease time of running test
+        exceptions_amount = 3
+        self.node.max_coredump_thread_exceptions = exceptions_amount
+        self.node.remoter.test_exception = NameError
+        self.node.remoter.exceptions_amount = exceptions_amount
+
+        self.node.start_backtrace_thread()
+
+        wait.wait_for(self.wait_for_all_exceptions_raised, timeout=60, exceptions_amount=exceptions_amount,
+                      text="Wait for all exceptions raised")
+
+        print("Wait for all exceptions raised")
+        traceback = wait.wait_for(self.find_last_event, step=0.1, timeout=10)
+
+        assert "NameError" == traceback, \
+               f"Expected exception 'Exception', got {traceback}"
+
+    def test_backtrace_thread_with_no_last_exception(self):
+        """
+        Test case when get backtraces function failed after an exception was raised 2 times and then continue
+        without failure.
+        get backtraces function should fail after an exception was raised 3 times. Because of just 2 exceptions will be
+        raised, expected that coredump thread continue to run.
+
+        """
+        exceptions_amount = 2
+        self.node.max_coredump_thread_exceptions = exceptions_amount+1
+        self.node.remoter.test_exception = ValueError
+        self.node.remoter.exceptions_amount = exceptions_amount
+
+        self.node.start_backtrace_thread()
+
+        wait.wait_for(self.wait_for_all_exceptions_raised, timeout=5, exceptions_amount=exceptions_amount)
```

Comment:
So, it used to organize a cycle with sleep inside?

Like:

```python
for _ in range(exceptions_amount):
    if self.node.remoter.run_number < min(exceptions_amount, 5):
        sleep(1)
```
???

Reply:
changed